dev/release/create-tarball.sh

```diff
@@ -80,11 +80,10 @@ echo ""
 echo "---------------------------------------------------------"
 cat <<MAIL
 To: dev@arrow.apache.org
```

Comment:
```suggestion
To: dev@datafusion.apache.org
```

Reply:
This mailing list isn't active yet. This PR is just to update references to the product name.

Reply:
There is a 24 hour waiting period between requesting a new mailing list and when it is created. I think the mailing lists will be active tomorrow